pkg/upgraders/osd/upgrader.go

```diff
@@ -635,6 +721,17 @@ func performUpgradeFailure(c client.Client, metricsClient metrics.Metrics, s sca
 	_, err := s.EnsureScaleDownNodes(c, nil, logger)
 	if err != nil {
 		logger.Error(err, "Failed to scale down the temporary upgrade machine when upgrade failed")
+		upgradeConfig.Status.History[0].Scaling.Failed = true
+		upgradeConfig.Status.History[0].Scaling.Dimension = "down"
+		if err = c.Status().Update(context.TODO(), upgradeConfig); err != nil {
+			return err
+		}
+		return err
```

Comment:
Seems like this could just be shortened to
```suggestion
		return c.Status().Update(context.TODO(), upgradeConfig)
```